core/sources/domains.move

```diff
@@ -427,22 +447,43 @@ module aptos_names::domains {
         if (option::is_some(&subdomain_name) && name_is_registerable(option::none(), domain_name)) {
             return false
         };
-        // Check to see if the domain is registered, or expired
-        let aptos_names = borrow_global<NameRegistryV1>(@aptos_names);
-        let name_record_key = create_name_record_key_v1(subdomain_name, domain_name);
-        !table::contains(&aptos_names.registry, name_record_key) || name_is_expired(subdomain_name, domain_name)
-    }
+        if (!name_is_registered(subdomain_name, domain_name)) {
+            // Name has never been registered. It is available.
+            return true
+        };
 
-    /// Returns true if the name is registered, and is expired.
-    /// If the name does not exist, raises an error
-    public fun name_is_expired(subdomain_name: Option<String>, domain_name: String): bool acquires NameRegistryV1 {
         let aptos_names = borrow_global<NameRegistryV1>(@aptos_names);
         let name_record_key = create_name_record_key_v1(subdomain_name, domain_name);
-        assert!(table::contains(&aptos_names.registry, name_record_key), error::not_found(ENAME_NOT_EXIST));
-
         let name_record = table::borrow(&aptos_names.registry, name_record_key);
         let (_property_version, expiration_time_sec, _target_address) = get_name_record_v1_props(name_record);
-        time_is_expired(expiration_time_sec)
+        let now = timestamp::now_seconds();
+        if (expiration_time_sec > now) {
+            // Name has not expired. It is not available.
+            return false
+        };
+
+        // Name has expired.
+        if (option::is_some(&subdomain_name)) {
+            // We don't check registration grace period for subdomains. Name is available.
+            return true
+        };
+
+        // Name has expired AND it is a domain. Check registration grace period.
+        let expired_for = now - expiration_time_sec;
+        return expired_for > config::reregistration_grace_sec()
+    }
+
+    /// Returns true if the name is not registered OR (is registered AND is expired)
+    public fun name_is_expired(subdomain_name: Option<String>, domain_name: String): bool acquires NameRegistryV1 {
+        if (!name_is_registered(subdomain_name, domain_name)) {
```

Comment:
This code is weird. This seems to only differ from name_is_registered in that it doesn't take into account the grace period. Why not create a common function that has a variable grace period and passes it 0 for name_is_expired and non-zero for name_is_registerable?